### PR TITLE
Upgrade deps: httpoison ~> 2.1, dialyxir ~> 1.4

### PR DESCRIPTION
**Pull request by jeremyowensboggs**
Upgrade the deps:
   :httpoison ~> 2.1
   :dialyxir ~> 1.4

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -26,9 +26,9 @@ defmodule Azurex.MixProject do
   # Run "mix help deps" to learn about dependencies.
   defp deps do
     [
-      {:dialyxir, "~> 1.1.0", only: [:dev, :test], runtime: false},
+      {:dialyxir, "~> 1.4", only: [:dev, :test], runtime: false},
       {:ex_doc, ">= 0.0.0", only: :dev, runtime: false},
-      {:httpoison, "~> 1.8"}
+      {:httpoison, "~> 2.1"}
```
Comment (pwcsquared) [inline mix.exs]: Thanks for opening this! Small suggestion to maintain compatibility with consumers of this library who might be stuck using httpoison 1.8 for other deps:

```suggestion
      {:httpoison, "~> 1.8 or ~> 2.2"}
```

**Review by pwcsquared — APPROVED** [review]
Looks good to me!

**Comment by pwcsquared** [comment]
@jakobht Any thoughts on merging this one? I'd love to be able to upgrade to httpoison > 2.0 in projects that use this library 🙂

**Review by jakobht — APPROVED** [review]
Looks good, sorry, the PRs got lost in all the GH spam 🫤

**Pull request merged**